pkg/machine/pull.go

```diff
@@ -68,6 +64,26 @@ func NewGenericDownloader(vmType, vmName, pullPath string) (DistributionDownload
 	return gd, nil
 }
 
+func supportedURL(path string) (url *url2.URL) {
+	getURL, err := url2.Parse(path)
+	if err != nil {
+		return nil
```

Comment:
This looks wrong.

Reply:
Ok I get what you are doing, might be nice to say that you are ignoring failure.

Reply:
ah, good point, created a follow-up pr for this: #17269